### PR TITLE
Timeout and memory

### PR DESCRIPTION
**Pull request by nelsonic**
@nikhilaravi as discussed in #41 & #35 this PR adds the ability to specify the `lambda_memory` and `lambda_timeout` in `package.json` so people don't have to do it *manually* in the UI. 👍 

Please review and let me know if anything unclear/missing. thanks!

### CODE REVIEW

File: lib/upload.js [hunk 1]
```diff
@@ -3,15 +3,29 @@ var AWS = require('aws-sdk');
...
 
-module.exports = function upload (callback) {
+/**
+ * upload does exacatly what you expect: uploads your lambda as a .zip to AWS!
+ * @param {Object} pkg - the package.json for the function you want to deploy
+ * this allows us to have a multi-lambda GitHub Repo or to deploy several Lambdas
+ * in one command by specifying the a different package.json for each of them.
+ * @param {Function} callback - called once the Lambda has been uploaded
+ */
+module.exports = function upload (pkg, callback) {
```
Comment (nikhilaravi) [inline lib/upload.js]: how does the user specify the package.json to use?

Comment (nelsonic) [inline lib/upload.js]: @nikhilaravi good question for now this is only used *internally* it will be exposed as a user feature in V4. 👍

**Comment by nelsonic** [comment]
@nikhilaravi let me know if anything else is unclear or needs to be changed. 👍

**Comment by nikhilaravi** [comment]
@nelsonic no all clear merging now!!

**Comment by nelsonic** [comment]
Thanks @nikhilaravi 👍

**Pull request merged**